### PR TITLE
Session recordings ingester

### PR DESCRIPTION
**Pull request by hazzadous**
This is a WIP branch for the work on Session Recordings ingestion revamp.

### CODE REVIEW

File: session-recordings/test/ingestion.test.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+            ],
+        })
+
+        const sessionRecording = await waitForSessionRecording(teamId, sessionId, eventUuid)
+
+        expect(sessionRecording.events.length).toBe(1)
+        expect(sessionRecording.events[0]).toMatchObject(JSON.parse(fullEventString))
+    })
+
+    // TODO: Handle this case and re-enable the test
```
Comment (hazzadous) [inline session-recordings/test/ingestion.test.ts:112]: @rcmarron what do you think the cases here are? I know we get some guarantees on this from Kafka, but I guess there might be other parts of the system that we need to consider?

Comment (hazzadous) [reply session-recordings/test/ingestion.test.ts:112]: For details on guarantees perhaps https://kafka.apache.org/documentation/#semantics is useful

**Comment by lancedouglas1** [comment]
Would adding a release-type allow this to be merged?  Is there anything I can help with? e.g. testing

**Comment by hazzadous** [comment]
> Would adding a release-type allow this to be merged? Is there anything I can help with? e.g. testing

@lancedouglas1 this has been put on hold for the time being as alternative solutions are discussed. Help with testing would be amazing though as well as any other input you can give. I gather you are on self hosted.

May I ask what you use case/interest in this is for?

**Comment by lancedouglas1** [comment]
> > Would adding a release-type allow this to be merged? Is there anything I can help with? e.g. testing
> 
> @lancedouglas1 this has been put on hold for the time being as alternative solutions are discussed. Help with testing would be amazing though as well as any other input you can give. I gather you are on self hosted.
> 
> May I ask what you use case/interest in this is for?

I'm currently on cloud hosted as it works well, except the retention of recordings.

My usecase has to do with retention lengths of recordings due to compliance.  Basically, it is very rarely that we have to go back and review a session recording to confirm user activity with 60 days of the actual activity.  It is almost always 90+ days later that other processes flag a session as requiring investigation.
Our alternative is to self host and just save indefinitely but that would require a much larger investment in posthog that is not yet approved.  Other things we're doing is putting more anomaly detection in our systems to close the window of time between events and investigations.

To be clear, we're not using recordings for security purposes, but rather we manage reservations for government owned properties, which means that we have many stakeholders with various interests in different types of citizenry-involved events that have happened in the past.

**Pull request closed without merging**